app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class LikesController < ApplicationController
+  def create
+    @like = Like.new(
+      user: current_user
+    )
+    set_likeable(params[:likeable_type], params[:likeable_id])
+    nil unless @like.save
+  end
+
+  def destroy
+    likeable_id = params[:likeable_id]
+    @like = Like.find_by(user_id: current_user.id, likeable_id:)
```

Comment:
If the user liked a post and a comment with the same ID, you could be deleting the incorrect one.